src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -250,14 +251,20 @@ CreateReferenceTable(Oid relationId)
  */
 static void
 ConvertToDistributedTable(Oid relationId, char *distributionColumnName,
-						  char distributionMethod, uint32 colocationId,
-						  char replicationModel)
+						  char distributionMethod, uint32 colocationId)
 {
 	Relation relation = NULL;
 	TupleDesc relationDesc = NULL;
 	char *relationName = NULL;
 	char relationKind = 0;
 	Var *distributionColumn = NULL;
+	char replicationModel = REPLICATION_MODEL_INVALID;
+
+	/* check global replication settings before continuing */
+	EnsureReplicationSettings(InvalidOid);
+
+	replicationModel = (distributionMethod == DISTRIBUTE_BY_NONE) ?
```

Comment:
Heh, I didn't know you could leave reactions on Pull Request comments, @ozgune, but I was like "I bet I know who would agree"…

I'll switch to an if even though I think the assignment RHS of `(bool) ? one_value : other_value` is so common an idiom as to be transparent.